docs/src/features/hooks.md

```diff
@@ -63,23 +64,118 @@ The function you provide into this hook must return either a `Model` or `QuerySe
+
+    === "components.py"
+
+        ```python
+        from idom import component
+        from django_idom.hooks import use_query
 
-    Due to Django's ORM design, database queries must be deferred using hooks. Otherwise, you will see a `SynchronousOnlyOperation` exception.
+        def example_query(value: int, other_value: bool = False):
+            ...
 
-    This may be resolved in a future version of Django containing an asynchronous ORM.
+        @component
+        def my_component():
+            query = use_query(
+                example_query,
+                123,
+                other_value=True,
+            )
+
+            ...
+        ```
 
-??? question "Why does the example `get_items` function return a `Model` or `QuerySet`?"
+??? question "Why does the example `get_items` function return `TodoItem.objects.all()`?"
 
-    This was a technical design decision to [based on Apollo](https://www.apollographql.com/docs/react/data/mutations/#usemutation-api), but ultimately helps avoid Django's `SynchronousOnlyOperation` exceptions.
+    This was a technical design decision to based on [Apollo's `useQuery` hook](https://www.apollographql.com/docs/react/data/queries/), but ultimately helps avoid Django's `SynchronousOnlyOperation` exceptions.
 
     The `use_query` hook ensures the provided `Model` or `QuerySet` executes all [deferred](https://docs.djangoproject.com/en/dev/ref/models/instances/#django.db.models.Model.get_deferred_fields)/[lazy queries](https://docs.djangoproject.com/en/dev/topics/db/queries/#querysets-are-lazy) safely prior to reaching your components.
 
-??? question "What is an "ORM"?"
+??? question "Can this hook be used for things other than the Django ORM?"
+
+    {% include-markdown "../../includes/orm.md" start="<!--orm-fetch-start-->" end="<!--orm-fetch-end-->" %}
+
+    However, if you...
+
+    1. Want to use this hook to defer IO intensive tasks to be computed in the background
+    2. Want to to utilize `use_query` with a different ORM
+
+    ... then you can disable all postprocessing behavior by modifying the `QueryOptions.postprocessor` parameter. In the example below, we will set the `postprocessor` to a function that takes one argument and returns `None`.
+
+    === "components.py"
+
+        ```python
+        from idom import component
+        from django_idom.types import QueryOptions
+        from django_idom.hooks import use_query
+
+        def execute_io_intensive_operation():
+            """This is an example query function that does something IO intensive."""
+            pass
 
-    A Python **Object Relational Mapper** is an API for your code to access a database.
+        @component
+        def todo_list():
+            # By setting the postprocessor to a function that takes one argument
+            # and returns None, we can disable postprocessing behavior.
+            query = use_query(
+                QueryOptions(postprocessor=lambda data: None),
```

Comment:
While I was trying to fix this, I discovered mypy does not like our type hint for postprocessor 


`Incompatible types in assignment (expression has type "Callable[[Union[_QuerySet[Any, Any], Model], bool, bool], None]", variable has type "Optional[Postprocessor]")`
```python
class Postprocessor(Protocol):
    def __call__(self, data: Any, **kwargs: Any) -> None:
        ...

@dataclass
class QueryOptions:
    """Configuration options that can be provided to `use_query`."""

    postprocessor: Postprocessor | None = django_query_postprocessor
```

Reply:
`**kwargs: Any` in the protocol says that post processor functions must accept arbitrary kwargs. That is, they must be a callable with a `**` param in them. My recommendation would be to just type the post processor as `Callable[..., None]`. It would be cool if you could do:

```python
P = ParamSpec("P")

class QueryOptions:
    postprocessor: Callable[Concatenate[Any, P], None]
    postprocessor_args: P.args
    postprocessor_kwargs: P.kwargs
```

But I don't think this is possible.

Reply:
I think you could do:
```python
class Postprocessor(Protocol):
    def __call__(self, data: Any, **kwargs: Any) -> None:
        ...
```

Since `(data: Any, kwarg_with_default: Any = ...) -> None` is technically compatible since the extra kwarg is optional.

Reply:
I am struggling to see the difference between your code block and what was typed above

Reply:
sorry. i corrected the code block